homeassistant/components/template/sensor.py

```diff
@@ -184,10 +191,15 @@ def name(self):
         """Return the name of the sensor."""
         return self._name
 
+    @property
+    def available(self) -> bool:
+        """Return the entity availability status."""
+        return bool(self._available)
+
     @property
     def state(self):
         """Return the state of the sensor."""
-        return self._state
+        return self._state if self.available else None
```

Comment:
There is no need for this. Home Assistant will not fetch `state` property if the entity is not available.